streaming/plugins/abci/v1/grpc.go

```diff
@@ -0,0 +1,108 @@
+package v1
+
+import (
+	"context"
+	"os"
+
+	"github.com/hashicorp/go-plugin"
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var _ storetypes.ABCIListener = (*GRPCClient)(nil)
+
+// GRPCClient is an implementation of the ABCIListener interface that talks over RPC.
+type GRPCClient struct {
+	client ABCIListenerServiceClient
+}
+
+func (m *GRPCClient) ListenBeginBlock(goCtx context.Context, req abci.RequestBeginBlock, res abci.ResponseBeginBlock) error {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	sm := ctx.StreamingManager()
+	request := &ListenBeginBlockRequest{Req: &req, Res: &res}
+	_, err := m.client.ListenBeginBlock(ctx, request)
+	if err != nil && sm.StopNodeOnErr {
+		ctx.Logger().Error("BeginBlock listening hook failed", "height", ctx.BlockHeight(), "err", err)
+		cleanupAndExit()
+	}
```

Comment:
```suggestion
	if err != nil {
		ctx.Logger().Error("BeginBlock listening hook failed", "height", ctx.BlockHeight(), "err", err)
		if sm.StopNodeOnErr {
			cleanupAndExit()
		}
	}
```

Similar for all the other spots like this. The error should be logged regardless of whether or not we're stopping the node.

Reply:
I had it that way because the error is already logged upstream in abci events when we're not stopping the stream. I didn't want double logging.